pkg/controller/bundle/bundle_unpacker_test.go

```diff
@@ -751,6 +756,523 @@ func TestConfigMapUnpacker(t *testing.T) {
+			description: "CatalogSourcePresent/JobPending/PodPending/BundlePending/WithContainerStatus",
+			fields: fields{
+				objs: []runtime.Object{
+					&corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      pathHash + "-pod",
+							Namespace: "ns-a",
+							Labels:    map[string]string{"job-name": pathHash},
+						},
+						Status: corev1.PodStatus{
+							Phase: corev1.PodPending,
+							InitContainerStatuses: []corev1.ContainerStatus{
+								{
+									Name:  "pull",
+									Ready: false,
+									State: corev1.ContainerState{
+										Waiting: &corev1.ContainerStateWaiting{
+											Reason:  "ErrImagePull",
+											Message: "pod pending for some reason",
+										},
+									},
+								},
+							},
+						},
+					},
+					&batchv1.Job{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      pathHash,
+							Namespace: "ns-a",
+							OwnerReferences: []metav1.OwnerReference{
+								{
+									APIVersion:         "v1",
+									Kind:               "ConfigMap",
+									Name:               pathHash,
+									Controller:         &blockOwnerDeletion,
+									BlockOwnerDeletion: &blockOwnerDeletion,
+								},
+							},
+						},
+						Spec: batchv1.JobSpec{
+							BackoffLimit: &backoffLimit,
+							Template: corev1.PodTemplateSpec{
+								ObjectMeta: metav1.ObjectMeta{
+									Name: pathHash,
+								},
+								Spec: corev1.PodSpec{
+									RestartPolicy: corev1.RestartPolicyNever,
+									Containers: []corev1.Container{
+										{
+											Name:    "extract",
+											Image:   opmImage,
+											Command: []string{"opm", "alpha", "bundle", "extract", "-m", "/bundle/", "-n", "ns-a", "-c", pathHash},
+											Env: []corev1.EnvVar{
+												{
+													Name:  configmap.EnvContainerImage,
+													Value: bundlePath,
+												},
+											},
+											VolumeMounts: []corev1.VolumeMount{
+												{
+													Name:      "bundle",
+													MountPath: "/bundle",
+												},
+											},
+											Resources: corev1.ResourceRequirements{
+												Requests: corev1.ResourceList{
+													corev1.ResourceCPU:    resource.MustParse("10m"),
+													corev1.ResourceMemory: resource.MustParse("50Mi"),
+												},
+											},
+										},
+									},
+									InitContainers: []corev1.Container{
+										{
+											Name:    "util",
+											Image:   utilImage,
+											Command: []string{"/bin/cp", "-Rv", "/bin/cpb", "/util/cpb"}, // Copy tooling for the bundle container to use
+											VolumeMounts: []corev1.VolumeMount{
+												{
+													Name:      "util",
+													MountPath: "/util",
+												},
+											},
+											Resources: corev1.ResourceRequirements{
+												Requests: corev1.ResourceList{
+													corev1.ResourceCPU:    resource.MustParse("10m"),
+													corev1.ResourceMemory: resource.MustParse("50Mi"),
+												},
+											},
+										},
+										{
```

Comment:
If this pod spec is identical in most test cases, can we factor it out?

Reply:
I don't think so. I tried to trim down the Job spec and the problem is that `func (c *ConfigMapUnpacker) UnpackBundle()`
will fail to ensure that the expected Job is present with the desired spec.
https://github.com/operator-framework/operator-lifecycle-manager/blob/a326d83ff42bda14a6b206675f54f092fcec815c/pkg/controller/bundle/bundle_unpacker.go#L411-L430

While testing this I've realized that we could end up with a nil job when we find and delete an unpack job that has the incorrect spec or ownerrefs.
https://github.com/operator-framework/operator-lifecycle-manager/blob/a326d83ff42bda14a6b206675f54f092fcec815c/pkg/controller/bundle/bundle_unpacker.go#L358-L362

So I'm going to update to also return and retry when `job == nil` as well otherwise we could see a nil dereference later on.